crates/rpc/rpc/src/result.rs

```diff
@@ -131,8 +132,38 @@ pub(crate) fn rpc_err(code: i32, msg: impl Into<String>, data: Option<&[u8]>) ->
     ))
 }
 
+pub(crate) trait ToRpcResultExt<T, Err> {
+    type Output;
+    type Error;
```

Comment:
we don't want this here.

this trait should be implemented manually for specific error types.